modules/electrophysiology_browser/php/electrophysiology_session.class.inc

```diff
@@ -0,0 +1,76 @@
+<?php declare(strict_types=1);
+
+/**
+ * This class features the code for the menu portion of the LORIS
+ * electrophysiology browser module.
+ *
+ * PHP version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Electrophysiology
+ * @author     Loris team <info-loris.mni@mcgill.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://github.com/aces/Loris
+ */
+namespace LORIS\electrophysiology_browser;
+/**
+ * Provides the PHP code for the menu filter for the electrophysiology browser
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Electrophysiology
+ * @author     Loris team <info-loris.mni@mcgill.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://github.com/aces/Loris
+ */
+
+class Electrophysiology_Session extends \NDB_Form
+{
+
+    public $skipTemplate = true; // stops from looking for a smarty template
+
+    /**
+     * Determine whether the user has permission to view this page
+     *
+     * @param \User $user The user whose access is being checked
+     *
+     * @return bool whether the user has access
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        $timePoint = \NDB_Factory::singleton()->timepoint(
+            intval($_REQUEST['sessionID'])
+        );
+
+        return ($user->hasPermission('electrophysiology_browser_view_allsites')
+                || ((in_array(
+                    $timePoint->getData('CenterID'),
+                    $user->getData('CenterIDs')
+                ))
```

Comment:
We should create a Utility function for this idiom. It's used in a lot of places and doesn't look very good in the code.

Reply:
sounds like a great plan to create a Utility function for this. Should this be done in a separate though since it would affect multiple modules? Let me know your thoughts on this.